android/src/main/java/com/ledger/reactnative/HttpClientImpl.java

```diff
@@ -55,8 +55,12 @@ public void run() {
                     byte[] body = request.getBody();
                     if (body.length > 0) {
                         connection.setRequestMethod( "POST" );
-                        connection.setRequestProperty( "Content-Type", "application/json");
-                        connection.setRequestProperty( "Content-Encoding", "UTF-8");
+                        if (!headers.containsKey("Content-Type")) {
```

Comment:
I’m not sure about that. This is a huge hidden side-effect that would basically prevent any usage of a HTTP client for anything that is not waiting JSON (I’m thinking protobuf, for instance). Why not asking the client to provide the right headers?

Reply:
it seems correct behavior of what libcore expected so far tho, we've been doing the exact same thing in Node.js side:
https://github.com/LedgerHQ/ledger-live-common/blob/313b85da993da3f91c6a8da52e45ec8f8ab84e34/src/libcore/platforms/nodejs.js#L155-L157

BTW providing a network implementation for libcore is really a gray area and not very clear on what libcore expects. similarly it's not clear what we assume to be a "success" vs an "error"


lot of diff of impl

Node.js

https://github.com/LedgerHQ/ledger-live-common/blob/313b85da993da3f91c6a8da52e45ec8f8ab84e34/src/libcore/platforms/nodejs.js#L138-L140

success errors and our need to remap the errors

iOS & Android

why do they fallback on "POST" when it's not provided 🤔 

https://github.com/LedgerHQ/lib-ledger-core-react-native-bindings/blob/master/ios/Sources/objc-impl/LGHttpClientImpl.m#L23-L27

https://github.com/LedgerHQ/lib-ledger-core-react-native-bindings/blob/master/android/src/main/java/com/ledger/reactnative/HttpClientImpl.java

Reply:
My concern is not about _what we’ve been doing for all this time_. It’s not because we’ve been doing something for years that it’s right. 🙂 If we have a class named `HTTPClient` and a function `POST`, we must **not** assume it will automatically do JSON requests.

I’m saying this because I wanna foresee our next move to _Ubinder_, which is very likely to require `protobuf` calls somehow.

Reply:
I agree better way for handling this would be to configure the http client on libcore side. The thing is when we do it on libcore (for Stellar for example), it gets override by the Android and iOS HttpClient implementation.
It requires to change all calls on libcore side though, I can create a bug ticket on libcore side, but I don't have the time to change all libcore calls and making sure that it doesn't break WD on live.

For Ubinder, it could be an issue if you have a very high level HTTP server which actually cares about HTTP headers, I don't think it will be the case.

Reply:
I agree with what's being said.

I suggest the plan:

- we accept this quick fix for the short term.
- we more clearly specified and document what is a "valid http client implementation" and work together on that. in order to eventually refine the libcore usage AND implementations. in the process of documenting that, I might have some usecase to raise (like the need for us to get the actual body out of a libcore error that is due to HTTP error – at the moment libcore just "eat" the error and we no longer access that body. that's what express in LLC-316)

Reply:
I’m fine with it if a Jira ticket is created. Feel free to put it in the `Housekeeping` epic I created a few days ago. 🙂

Reply:
I create LLC-555 just for you @phaazon  😛.

Reply:
Yes this will be part of the first wave of `Housekeeping` from our side.
But I agree that we can accept this as a patch that will be reviewed during that same wave.
@pollastri-pierre could you please add a ticket link describing the incentive behind that patch (e.g. the exact use case from coin integration side) ? 
This will help us when we will address that problem ! 
Thanks for the patch !